packages/eslint-config-godaddy-minimal/index.js

```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  extends: [
+    require('./extends')('eslint-config-godaddy-react-flow')
+  ],
+  rules: {
+    'complexity': 0,
+    'consistent-return': 0,
+    'id-length': 0,
+    'indent': [2, 2, {
+      VariableDeclarator: { var: 2, let: 2, const: 3 },
+      flatTernaryExpressions: true,
+      MemberExpression: 'off'
+    }],
+    'max-params': 0,
```

Comment:
Because JavaScript does not have named functions, function calls with too many parameters become unreadable. I have seen far too many code bases where huge parameter lists make reability a nightmare. Also, it's generally bad code smell that a function has too many responsibilities.

"option objects" are far more readable if passing a lot of data is required, since all the pieces of data are named.

Reply:
"option objects" are an abstraction built around "I needed all of this data in one place, so I shoved them together". This is not a good abstraction, or something we should encourage. Options objects are also not a general solution, because they complicate type checking and the use of higher-order functions.

If there's a large number of arguments to a function (a function which is exposed to callers, not one which has different sets of default arguments bound to it to produce simpler functions) then the solution is to build better domain abstractions to combine the function's arguments, or more frequently, to divide the function up, separating its responsibilities. Or to do what I just mentioned, and export multiple functions, each of which are the result of binding different default arguments to the main one. There's a lot of wiggle room here, and these are tradeoffs that should be discussed in code review. Making this a lint rule encourages people to bury the problem.

Reply:
The fact that this sometimes is helpful and sometimes is harmful is a sign that a minimal set of rules should leave it out, and let teams which want it enable it in their project overrides. If I thought that this rule was harmful 100% of the time I'd be pushing to disable it in the main rule set.

Reply:
I agree with jpage. +1 for options object.

Reply:
This may be a case of loosening the restriction a bit. 4 params is extreme, 6 would be better and still get the point across. 

I don't agree that having an object parameter with a ton of properties is any more readable or less of a code smell than a parameter list of the same number. Not saying a lot of params/properties is a good thing, but not understanding how wrapping them in an object makes it any better, plus wrapping forces you to have to change the caller's signatures wherever they may be if a new parameter is an enhancement to the function.